### PR TITLE
Save image link

### PR DESCRIPTION
**Pull request by aimsecond**
Finished Issue#2 with following features created:

-  The landing page links to the image link submission form.
- The link is entered through a form.
- After the form is saved, the link is persisted in the database.
- After the form is submitted, the user is redirected to a page displaying
the image.
- Users cannot successfully save an image with an invalid URL.
- An error message is associated with the appropriate input field on
failure.

What is a valid image URL?
My valid image URL definition: 
- [ ] The protocal(http or https) and Hostname can be retrieved from the target url.
- [ ] A successful respond can be acquired from the target url
- [ ] The content-type from the response should be started with 'image'

### CODE REVIEW

**Comment by aimsecond** [comment]
[Ropes project issue#2](https://trello.com/c/FP6PTheL/2065-ropes-project-issue2)

File: app/models/picture.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Picture < ApplicationRecord
```
Comment (gauravc88) [inline app/models/picture.rb]: The picture model should include the name and link of the picture and validations for them

File: app/controllers/pictures_controller.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  def new; end
+
+  def create
+    img_url = picture_params[:link]
+    if remote_image_exists?(img_url_valid?(img_url))
+      @picture = Picture.new(picture_params)
+      @picture.save
+      redirect_to @picture
+    else
+      flash.now[:info] = 'Incorrect Image URL!'
```
Comment (gauravc88) [inline app/controllers/pictures_controller.rb]: Lets change this to a flash error instead of info

File: app/controllers/pictures_controller.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+require 'uri'
+
+class PicturesController < ApplicationController
+  def new; end
+
+  def create
+    img_url = picture_params[:link]
+    if remote_image_exists?(img_url_valid?(img_url))
+      @picture = Picture.new(picture_params)
+      @picture.save
```
Comment (gauravc88) [inline app/controllers/pictures_controller.rb]: There might be one more if required if the picture validation fails

File: test/controllers/pictures_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PicturesControllerTest < ActionDispatch::IntegrationTest
```
Comment (gauravc88) [inline test/controllers/pictures_controller_test.rb]: Lets add tests for the code here

File: test/models/picture_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PictureTest < ActiveSupport::TestCase
```
Comment (gauravc88) [inline test/models/picture_test.rb]: Once fields are added to the picture class then tests should be added here

**Comment by aimsecond** [comment]
test review msg

**Pull request merged**